src/Symfony/Component/Messenger/Transport/Schedule/Schedule.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Schedule;
+
+use Symfony\Component\Clock\ClockInterface;
+
+/**
+ * @internal used to bypass a different evaluation order issue in class constant initializers
+ *
+ * @see https://wiki.php.net/rfc/new_in_initializers#unsupported_positions
+ */
+const _SCHEDULE_FAR_FUTURE_ = new \DateTimeImmutable('3000-01-01');
+
+class Schedule implements \IteratorAggregate
+{
+    public const FAR_FUTURE = _SCHEDULE_FAR_FUTURE_;
```

Comment:
this could be private: this variable is publicly used in one case: when the next run is after the end of the period. in that case the name is wrong : it's not "far in the future" but should be "never". 
And the scheduler could return null to tell the transport "don't call me again".

Reply:
Personally, I hate returning nulls as special case. But that constant seems more confusing, so I removed it.